File: src/test/java/com/ject/studytrip/trip/helper/DailyGoalTestHelper.java
package com.ject.studytrip.trip.helper;

import com.ject.studytrip.trip.domain.model.DailyGoal;
import com.ject.studytrip.trip.domain.model.Trip;
import com.ject.studytrip.trip.domain.repository.DailyGoalRepository;
import com.ject.studytrip.trip.fixture.DailyGoalFixture;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class DailyGoalTestHelper {

    @Autowired private DailyGoalRepository dailyGoalRepository;

    public DailyGoal saveDailyGoal(Trip trip) {
        DailyGoal dailyGoal = DailyGoalFixture.createDailyGoal(trip);
        return dailyGoalRepository.save(dailyGoal);
    }

    public DailyGoal saveDeletedDailyGoal(Trip trip) {
        DailyGoal dailyGoal = DailyGoalFixture.createDailyGoal(trip);
        dailyGoal.updateDeletedAt();

        return dailyGoalRepository.save(dailyGoal);
    }
}

File: src/test/kotlin/com/ject/studytrip/trip/helper/DailyGoalTestHelper.kt
package com.ject.studytrip.trip.helper

import com.ject.studytrip.trip.domain.model.DailyGoal
import com.ject.studytrip.trip.domain.model.Trip
import com.ject.studytrip.trip.domain.repository.DailyGoalRepository
import com.ject.studytrip.trip.fixture.DailyGoalFixture
import org.springframework.stereotype.Component

@Component
class DailyGoalTestHelper(
    private val dailyGoalRepository: DailyGoalRepository,
) {
    fun saveDailyGoal(trip: Trip): DailyGoal = dailyGoalRepository.save(DailyGoalFixture(trip).create())

    fun saveDeletedDailyGoal(trip: Trip): DailyGoal =
        dailyGoalRepository.save(DailyGoalFixture(trip).create().also { it.updateDeletedAt() })
}
